Automattic/studio
# fix(sync): handle reverted WoA sites

## Related issues

- Fixes https://github.com/Automattic/dotcom-forge/issues/9374

## Proposed Changes
There is a case where user connected WoA site with Business plan and then reverted it, so it becomes with errors as "Unsupported plan" or "Please enable hosting features".
With this PR we handle only reverting sites. Deleted sites will be handled in separate PR, to simplify review, since there is nuance - ATM [we don't load deleted sites](https://github.com/Automattic/studio/blob/trunk/src/hooks/use-fetch-wpcom-sites.tsx#L109), so necessary to think and maybe make some refactoring of existing code.

## Testing Instructions
1. Checkout to this branch and run `STUDIO_SITE_SYNC=true npm start`
2. Open wordpress.com and purchase two sites with Business plan
3. Open Studio and connect one of them
4. Open wordpress.com/me/purchases and cancel Business plan subscription for previously connected site
5. Open Studio and assert that you see error:<br /><img width="847" alt="Screenshot 2024-11-20 at 12 09 07" src="https://github.com/user-attachments/assets/a742f5ef-2439-407b-afa6-2c2d38cbd808">
6. Click on "Reconnect" button
7. Assert that you see dialog for connecting sites
8. Find your unsupported site
9. Asser that it has "Unsupported plan"<br /><img width="654" alt="Screenshot 2024-11-20 at 12 10 56" src="https://github.com/user-attachments/assets/8c74a87b-14cb-4522-b0ef-c182b5cd023b">
10. Reset search ("x" button on the right side of search field)
11. Find the second site, which was bought on the second step in this Testing instructions, select and click "Conenct button"
12. Assert that after reconnecting the old connection disappeared and the new connection appeared into Sync tab
